File: java/previous/resistor-color/src/test/java/ResistorColorTest.java
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeEach;

import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class ResistorColorTest {

    private ResistorColor resistorColor;

    @BeforeEach
    public void setup() {
        resistorColor = new ResistorColor();
    }

    @Test
    public void testBlackColorCode() {
        assertEquals(0, resistorColor.colorCode("black"));
    }

    @Test
    public void testBrownColorCode() {
        assertEquals(1, resistorColor.colorCode("brown"));
    }

    @Test
    public void testRedColorCode() {
        assertEquals(2, resistorColor.colorCode("red"));
    }

    @Test
    public void testOrangeColorCode() {
        assertEquals(3, resistorColor.colorCode("orange"));
    }

    @Test
    public void testYellowColorCode() {
        assertEquals(4, resistorColor.colorCode("yellow"));
    }

    @Test
    public void testGreenColorCode() {
        assertEquals(5, resistorColor.colorCode("green"));
    }

    @Test
    public void testBlueColorCode() {
        assertEquals(6, resistorColor.colorCode("blue"));
    }

    @Test
    public void testVioletColorCode() {
        assertEquals(7, resistorColor.colorCode("violet"));
    }

    @Test
    public void testGreyColorCode() {
        assertEquals(8, resistorColor.colorCode("grey"));
    }

    @Test
    public void testWhiteColorCode() {
        assertEquals(9, resistorColor.colorCode("white"));
    }

    @Test
    public void testColorsCodeBlackBrownRed() {
        assertEquals("012", resistorColor.colorsCode("blackbrownred"));
    }

    @Test
    public void testColorsCodeVioletGreyWhite() {
        assertEquals("789", resistorColor.colorsCode("violetgreywhite"));
    }

    @Test
    public void testColorsCodeAllColor() {
        assertEquals("0123456789", resistorColor.colorsCode("blackbrownredorangeyellowgreenbluevioletgreywhite"));
    }

    
    @Test
    public void testColors() {
        List<String> colors = Arrays.asList("black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white");

        assertEquals(colors, resistorColor.colors());
    }
}

File: javascript/previous/resistor-color/resistor-color.test.js
const { resistorColor, resistorColors } = require( "./resistor-color.js");

describe('Resistor Color Tests', () => {
  test('black to be 0', () => {
    expect(resistorColor('black')).toBe(0);
  });

  test('brown to be 1', () => {
    expect(resistorColor('brown')).toBe(1);
  });

  test('red to be 2', () => {
    expect(resistorColor('red')).toBe(2);
  });

  test('orange to be 3', () => {
    expect(resistorColor('orange')).toBe(3);
  });

  test('yellow to be 4', () => {
    expect(resistorColor('yellow')).toBe(4);
  });

  test('green to be 5', () => {
    expect(resistorColor('green')).toBe(5);
  });

  test('blue to be 6', () => {
    expect(resistorColor('blue')).toBe(6);
  });

  test('violet to be 7', () => {
    expect(resistorColor('violet')).toBe(7);
  });

  test('grey to be 8', () => {
    expect(resistorColor('grey')).toBe(8);
  });

  test('white to be 9', () => {
    expect(resistorColor('white')).toBe(9);
  });

  test('BlackBrownRed to be 012', () => {
    expect(resistorColors('blackbrownred')).toBe('012');
  });

  test('BlackBrownRed to be 789', () => {
    expect(resistorColors('violetgreywhite')).toBe('789');
  });

  test('All colors to be 0123456789', () => {
    expect(resistorColors('blackbrownredorangeyellowgreenbluevioletgreywhite')).toBe('0123456789');
  });
});
 